compiler/rustc_hir_analysis/src/hir_ty_lowering/cmse.rs

```diff
@@ -67,13 +70,13 @@ pub(crate) fn validate_cmse_abi<'tcx>(
 /// Returns whether the inputs will fit into the available registers
 fn is_valid_cmse_inputs<'tcx>(
     tcx: TyCtxt<'tcx>,
-    fn_sig: ty::PolyFnSig<'tcx>,
+    fn_sig: ty::FnSig<'tcx>,
```

Comment:
One tweak -- can you change this back to `PolyFnSig` and do the binder instantiate call inside this fn?